File: Baekjoon/prime_1929.java
// https://www.acmicpc.net/problem/1929
// BOJ_1929_소수 구하기
import java.util.*;
public class prime_1929 {

	public static void getPrime(int m, int n, ArrayList<Integer> prime) {
		prime.add(2);
    for (int i = 2; i <= n; i++) {
			for (int j = 0; prime.size() > j; j++) {
				if (i % prime.get(j) == 0) break; // 소수가 아닌 경우 pass
				if (j + 1 == prime.size()) // 소수일 때
					prime.add(i);
			}
		}
		for (Integer result : prime) {
      if(result >= m)
			System.out.println(result);
		}
	}
	public static void main(String[] args) {
    Scanner in = new Scanner(System.in);
		ArrayList<Integer> prime = new ArrayList<Integer>();
    int m = in.nextInt();
    int n = in.nextInt();
		getPrime(m, n, prime);
	}
}

File: Baekjoon/prime_1929.cpp
// https://www.acmicpc.net/problem/1929
// BOJ_1929_소수 구하기

#include <stdio.h>
#include <math.h>
int main() {
    int m, n, i, j, chk;
    scanf("%d %d", &m, &n);
    if(n == 1) return 0;
    if(m <= 2) { printf("2\n"); m = 2; }
    for(i = m; i <= n; i++) {
        if(i % 2 == 0) continue;
        chk = 0;
        for(j = 2; j <= sqrt(i); j++){
            if(i % j == 0){
                chk = 1;
                break;
            }
        }
        if(chk == 0) printf("%d\n", i);
    }
    return 0;
}
